### PR TITLE
Implement Ord and PartialOrd for UniCase<AsRef<str>>

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

**Comment by seanmonstar** [comment]
Implementing `Ord` makes sense. I see there isn't a `cmp_ignore_case` method for `AsciiExt`. Could we use `memicmp` for this on all platforms?

**Comment by frewsxcv** [comment]
I'm not familiar with `memicmp` with Rust. Can you point me in the direction to learn more about how to move forward with it?

File: src/lib.rs [hunk 1]
```diff
@@ -41,6 +42,20 @@ impl<S> DerefMut for UniCase<S> {
...
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl<T: AsRef<str>> Ord for UniCase<T> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let self_chars = self.as_ref().chars().map(|c| c.to_ascii_lowercase());
+        let other_chars = other.as_ref().chars().map(|c| c.to_ascii_lowercase());
+        self_chars.cmp(other_chars)
```
Comment (seanmonstar) [inline src/lib.rs]: It seems this is using an unstable feature, so the tests don't pass on stable.

Comment (frewsxcv) [inline src/lib.rs]: Or I could wait until Thursday :)

https://github.com/rust-lang/rust/pull/29254

**Comment by frewsxcv** [comment]
Compiles on stable now :)

**Comment by seanmonstar** [comment]
Woo!

**Pull request merged**